native/src/core/su/su.cpp

```diff
@@ -224,7 +219,7 @@ int su_client_main(int argc, char *argv[]) {
     }
 
     // Determine which one of our streams are attached to a TTY
-    interactive |= req.command.empty();
+    interactive |= req.command.size() == 1;
```

Comment:
I recommend keeping `--shell` separated and leaving this `empty()`. Otherwise, `su some_commend` still errorously allocates tty (thus cannot `^Z`).

Reply:
Tested `su sleep 20` will not allocate tty

Reply:
I mean, only one command. For example, `su ls`.

Reply:
Tested `su ls` will not allocate tty